src/internal.c

```diff
@@ -35500,6 +35479,311 @@ static int DoSessionTicket(WOLFSSL* ssl, const byte* input, word32* inOutIdx,
+            case ed448_sa_algo:
+            {
+                ssl->buffers.keyType = ed448_sa_algo;
+                ret = DecodePrivateKey(ssl, &keySz);
+                /* worst case estimate */
+                args->tmpSigSz = ED448_SIG_SIZE;
+                break;
+            }
+        #endif /* HAVE_ED448 */
+            default:
+                ret = ALGO_ID_E;  /* unsupported type */
+        } /* switch(ssl->specs.sig_algo) */
+        return ret;
+    }
+    #endif /*HAVE_ECC || (HAVE_CURVE25519 || HAVE_CURVE448) &&
+               (HAVE_ED25519 || HAVE_ED448 || !NO_RSA)*/
+
+    #if !defined(NO_DH)\
+         && (!defined(NO_PSK) || !defined(NO_RSA) || \
+                             (defined(HAVE_ANON) && !defined(WOLFSSL_NO_TLS12)))
+    static void LengthPlusPGPub(WOLFSSL* ssl, SskeArgs* args)
+    {
+        args->length += LENGTH_SZ * 3 + /* p, g, pub */
+                 ssl->buffers.serverDH_P.length +
+                 ssl->buffers.serverDH_G.length +
+                 ssl->buffers.serverDH_Pub.length;
+    }
+
+    static void AddPGPub(WOLFSSL* ssl, SskeArgs* args)
+    {
+        /* add p, g, pub */
+        c16toa((word16)ssl->buffers.serverDH_P.length,
+                                    args->output + args->idx);
+        args->idx += LENGTH_SZ;
+        XMEMCPY(args->output + args->idx,
+                              ssl->buffers.serverDH_P.buffer,
+                              ssl->buffers.serverDH_P.length);
+        args->idx += ssl->buffers.serverDH_P.length;
+
+        /*  g */
+        c16toa((word16)ssl->buffers.serverDH_G.length,
+                                    args->output + args->idx);
+        args->idx += LENGTH_SZ;
+        XMEMCPY(args->output + args->idx,
+                              ssl->buffers.serverDH_G.buffer,
+                              ssl->buffers.serverDH_G.length);
+        args->idx += ssl->buffers.serverDH_G.length;
+
+        /*  pub */
+        c16toa((word16)ssl->buffers.serverDH_Pub.length,
+                                    args->output + args->idx);
+        args->idx += LENGTH_SZ;
+        XMEMCPY(args->output + args->idx,
+                              ssl->buffers.serverDH_Pub.buffer,
+                              ssl->buffers.serverDH_Pub.length);
+        args->idx += ssl->buffers.serverDH_Pub.length;
+    }
+    #endif
+
+    /*
+     * this #if mess is a case for relaxing the requirement that
+     * static functions that are defined must always be used.
+     *
+     * as it is, we have to be careful to only define the function
+     * when it is used, which is the || of all the cases below
+     *
+     */
+#if (!defined(NO_DH) && (!defined(NO_RSA) || \
+     (defined(HAVE_ANON) && !defined(WOLFSSL_NO_TLS12)))) || \
+    defined(HAVE_ECC) || \
+    ((defined(HAVE_CURVE25519) || defined(HAVE_CURVE448)) && \
+     (defined(HAVE_ED25519) || defined(HAVE_ED448) || !defined(NO_RSA)))
+    static int GetHashType(WOLFSSL* ssl, SskeArgs* args,
```

Comment:
You have an encoding of signature going on here which doesn't align with the function getting the hash type.

Change function name to something different.